Instagram/Models/Comment.m

```diff
@@ -0,0 +1,35 @@
+//
+//  Comment.m
+//  Instagram
+//
+//  Created by Megan Miller on 6/28/22.
+//
+
+#import "Comment.h"
+
+@implementation Comment
+
+@dynamic commentID;
+@dynamic userID;
+@dynamic author;
+@dynamic text;
+    
++ (nonnull NSString *)parseClassName {
+    return @"Comment";
+}
+
++ (void)postComment:(NSString * _Nullable)text forPost:(Post *)post
+     withCompletion:(PFBooleanResultBlock _Nullable)completion {
+    
+    Comment *newComment = [Comment new];
+    newComment.author = [PFUser currentUser];
+    newComment.text = text;
+    
+    [newComment saveInBackgroundWithBlock:completion];
+    
+    [post incrementKey:@"commentCount"];
+    [post addObject:newComment forKey:@"comments"];
+    [post saveInBackgroundWithBlock:completion];
+}
```

Comment:
All of that said, my suggestion would be to add this as an instance method to `Post` and pass in a `Comment`. 

```
- (void)addComment:(Comment *)comment completion:(PFBooleanResultBlock _Nullable)completion {
    [newComment saveInBackgroundWithBlock:nil];

    [post incrementKey:@"commentCount"];
    [post addObject:newComment forKey:@"comments"];
    [post saveInBackgroundWithBlock:completion];
}
```

That way at the call site it would look like

```
Comment *newComment = [Comment new];
newComment.author = [PFUser currentUser];
newComment.text = text;
[post addComment:newComment];
```